File: src/0876.middle-of-the-linked-list.908/solution.ts
class ListNode {
    val: number
    next: ListNode | null
    constructor (val?: number, next?: ListNode | null) {
        this.val = (val === undefined ? 0 : val);
        this.next = (next === undefined ? null : next);
    }
}

function middleNode (head: ListNode | null): ListNode | null {
    let fast:ListNode|null = head;
    let slow:ListNode|null = head;
    while (fast && fast.next) {
        fast = fast.next.next;
        slow = (slow as ListNode).next;
    }
    return slow;
}

File: src/0876.middle-of-the-linked-list.908/solution.go
package middleofthelinkedlist908

type ListNode struct {
	Val  int
	Next *ListNode
}

func middleNode(head *ListNode) *ListNode {
	slow := head
	fast := head
	for fast != nil && fast.Next != nil {
		fast = fast.Next.Next
		slow = slow.Next
	}
	return slow
}
